### PR TITLE
Creating a role for managing only profiles

### PR DESCRIPTION
**Pull request by amir-budaychiev**
*Adding a role moderator to the fake task
*Settings an ability file
*Settings the controllers

### CODE REVIEW

**Comment by amir-budaychiev** [comment]
@mod It's a good question. There was a problem related with dashboard controller when user authorized as compliance officer, action index gave an error. So, It was one of more solutions. I had to split logic to two controllers. But yeah, we have more ways to do that...let's check new solution.

**Comment by mod** [comment]
@amir-budaychiev maybe you remove dashboard and we display another controller like account list.

**Comment by amir-budaychiev** [comment]
@mod Sure thing.

**Comment by amir-budaychiev** [comment]
@mod @gfedorenko guys, can we merge that PR? Or I need to do more changes?

**Comment by mkuvshynnikov** [comment]
@amir-budaychiev  Can you do not display tabs that disabled  for the role "compliance"
cx3

**Comment by amir-budaychiev** [comment]
@mkuvshynnikov it's done

**Comment by mod** [comment]
Please continue implementation at controller level

File: app/helpers/view_helper.rb [hunk 1]
```diff
@@ -15,4 +15,15 @@ def badge_by_state(state)
...
 
+  def badge_by_role(role)
+    case role
+      when 'admin'
+        'badge-success'
+      when 'compliance'
+        'badge-warning'
+      when 'member'
+        'badge-info'
+    end
```
Comment (mitjok) [inline app/helpers/view_helper.rb]: It's always a good idea to catch `no role`, `new role` with `else` close in `case` statement, imo

File: app/models/ability.rb [hunk 2]
```diff
@@ -34,6 +34,11 @@ def initialize(account)
     # See the wiki for details:
     # https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities
 
-    can :manage, :all if account&.role&.admin?
+    case account.role
+      when 'admin'
+        can :manage, :all
+      when 'compliance'
+        can :manage,  [Profile]
+    end
```
Comment (mitjok) [inline app/models/ability.rb]: It's always a good idea to catch `no role`, `new role` with `else` close in `case` statement, imo

File: app/controllers/admin/module_controller.rb [hunk 3]
```diff
@@ -1,25 +1,13 @@
...
 
 module Admin
-  class ModuleController < ApplicationController
-    layout 'admin'
-
-    alias current_user current_account # CanCanCan expects current_user.
-
-    rescue_from CanCan::AccessDenied, with: :redirect_to_index
-
+  class ModuleController < BaseController
```
Comment (mitjok) [inline app/controllers/admin/module_controller.rb]: @amir-budaychiev what is the purpose of the `ModuleController`?

File: lib/tasks/db_load_fake.rake [hunk 4]
```diff
@@ -7,6 +7,9 @@ namespace :db do
       account = Account.create(email: 'admin@gmail.com', password: '123123', role: 'admin', confirmed_at: Faker::Time.between(2.days.ago, Date.today))
       Profile.create(account: account, first_name: Faker::Name.first_name, last_name: Faker::Name.last_name, country: Faker::Address.country, dob: Faker::Date.between(25.years.ago, 10.years.ago), address: Faker::Address.street_address, city: Faker::Address.city)
 
+      compliance = Account.create(email: 'compliance@gmail.com', password: '123123', role: 'compliance', confirmed_at: Faker::Time.between(2.days.ago, Date.today))
```
[the remark this answers is not in the document]
Comment (amir-budaychiev) [reply lib/tasks/db_load_fake.rake]: @mitjok all clear

File: app/helpers/view_helper.rb (hunk 1 above)
Comment (amir-budaychiev) [reply app/helpers/view_helper.rb]: @mitjok do you mean I need add 'else' condition?

File: app/models/ability.rb (hunk 2 above)
Comment (amir-budaychiev) [reply app/models/ability.rb]: @mitjok do you mean I need add 'else' condition?

File: app/controllers/admin/module_controller.rb (hunk 3 above)
Comment (amir-budaychiev) [reply app/controllers/admin/module_controller.rb]: @mitjok it's a good question, but I did a couple changes here, I didn't create that controller, you have to better ask the one who created @yivo 

but, as I understand - this is created for the inheritance of common characteristics admin's controllers.

**Comment by amir-budaychiev** [comment]
@mod @mitjok continuation in PR #193

**Pull request merged**